datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -372,6 +377,40 @@ pub fn date_part(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     })
 }
 
+fn extract_epoch<T>(array: &PrimitiveArray<T>) -> Result<Int32Array>
+where
+    T: ArrowTemporalType + ArrowNumericType,
+    i64: std::convert::From<T::Native>,
+{
+    let mut b = Int32Builder::new(array.len());
```

Comment:
Note: In terms of performance (and readability), it is better to avoid the builder/`value` etc. and work directly on the values (check the arrow-rs kernels for examples).
Ok to leave this for a future PR if you like 👍